wordpress-mobile/WordPress-iOS
# Avoid tracking widget events in the preview

Fixes #NA

This PR fixes an issue that was causing widget install events to be sent when the widget were only previewed and not actually installed.

To test:
- Pre-requisites:
    - It's recommended to use a simulator
    - To avoid waiting longer times, in `SiteListProvider` you can tweak:
        - `refreshInterval = 5`
        - `minElapsedTimeToRefresh = 1`
        - In the simulator settings, enable "WidgetKit developer mode"

- Test execution:
    - Build/run the app, then background it
    - in Xcode, Debug/Attach to process and select `WordPressStatsWidgets`
    - Do a variety of widget operations (preview, install, change site, delete, etc)
    - Use breakpoints in `Tracks+StatsWidgets.swift to make sure that:
        - No events are sent when in preview mode
        - events are only sent if a widget is installed, and not when the site in the same widget is changed
        - Wait approx 5 minutes so that iOS triggers a widget update, and make sure no events are sent in that case
    - The method that sends the tracks is `trackExtensionEvent`

PR submission checklist:
